### PR TITLE
Cut Layout Sheet

### PR DESCRIPTION
**Pull request by alzatin**

### CODE REVIEW

File: api/v1-bundle/maslowWorker.js [hunk 1]
```diff
@@ -46,16 +46,40 @@ const agent = async ({ ask, question }) => {
       case 'getLayoutSvgs':
         // Extract shapes
         var items = api.Shape.fromGeometry(values[0]).toItems();
+        var shiftedItems = [];
+        var spacing = values[1];
```
Comment (BarbourSmith) [inline api/v1-bundle/maslowWorker.js]: I've been told it is better form to use 'const' for things which won't change like the spacing. Not sure it's a big deal, but trying to be helpful style wise

File: api/v1-bundle/maslowWorker.js [hunk 2]
```diff
@@ -46,16 +46,40 @@ const agent = async ({ ask, question }) => {
...
+        var spacingY = 3;
+        var sheetX = values[2];
+        var sheetY = values[3];
+        
+        var lengths =[];
+        var widths =[];
+        items.forEach(item => {
```
Comment (BarbourSmith) [inline api/v1-bundle/maslowWorker.js]: I see why you need to do this twice. Let's not worry about it too much unless performance becomes an issue

File: api/v1-bundle/maslowWorker.js [hunk 3]
```diff
@@ -46,16 +46,40 @@ const agent = async ({ ask, question }) => {
...
+        var sheetX = values[2];
+        var sheetY = values[3];
+        
+        var lengths =[];
+        var widths =[];
+        items.forEach(item => { 
+          var flattened = item.flat();
+          var length = flattened.measureBoundingBox()[1][1] - flattened.measureBoundingBox()[0][1];
+          var width = flattened.measureBoundingBox()[1][0] - flattened.measureBoundingBox()[0][0];     
+          lengths.push(length);
```
Comment (BarbourSmith) [inline api/v1-bundle/maslowWorker.js]: This works fine, another option you see more often would be to have a variable called `maxLength` and then do `maxLength = Math.max(maxLength, length)`  It's basically doing the same thing.

File: api/v1-bundle/maslowWorker.js (hunk 2 above)
Comment (BarbourSmith) [reply api/v1-bundle/maslowWorker.js]: The only way I can see we could do this faster in terms of performance would be to do an initial loop which generates a new array of shapes called flatItems which is basically all the items flattened. Also items was a bad variable name choice on my part...sorry

File: api/v1-bundle/maslowWorker.js (hunk 1 above)
Comment (alzatin) [reply api/v1-bundle/maslowWorker.js]: yeah, i feel like it is still pretty unclear to me the essential difference of let, var and const

File: api/v1-bundle/maslowWorker.js (hunk 3 above)
Comment (alzatin) [reply api/v1-bundle/maslowWorker.js]: Would you rather it be this way?

**Comment by alzatin** [comment]
i changed it so it  would make an array of flat items instead of doing it twice

**Comment by BarbourSmith** [comment]
Nice work!

**Pull request merged**